modules/mapbox/src/deck-utils.js

```diff
@@ -88,9 +94,12 @@ export function drawLayer(deck, map, layer) {
   if (!deck.layerManager) {
     return;
   }
+
   deck._drawLayers('mapbox-repaint', {
-    viewports: [currentViewport],
-    layerFilter: ({layer: deckLayer}) => layer.id === deckLayer.id,
+    viewports: applyViewport(deck, map, currentViewport),
+    layerFilter: props =>
+      (deck.props.layerFilter && deck.props.layerFilter(props)) ||
+      (props.viewport.id === 'mapbox' && layer.id === props.layer.id),
```

Comment:
The two filters must both pass. This function is called once for every Mapbox Layer.

```js
params =>
      (!deck.props.layerFilter || deck.props.layerFilter(params)) &&
      (params.viewport.id === 'mapbox' && layer.id === params .layer.id),
```